src/time_tracker/invoices/core.clj

```diff
@@ -0,0 +1,56 @@
+(ns time-tracker.invoices.core
+  (:require [clojure.data.csv :as csv]
+            [time-tracker.timers.core :as timers-core]))
+
+(defn empty-time-map
+  "Returns a map of {user-id {project-id 0}}
+  for all users and projects. The last entry is the number of hours.
+  There must be at least one user and one project."
+  [users projects]
+  (->> (for [user-id (keys users)
+             project-id (keys projects)]
+         {user-id {project-id 0}})
+       (apply merge-with merge)))
+
+(defn- seconds->hours
+  [seconds]
+  (double (/ seconds 3600)))
+
+(defn add-hours
+  "Adds the hours logged against the timer entry
+  to the time map."
+  [time-map {:keys [project-id app-user-id] :as timer}]
+  (update-in time-map
+             [app-user-id project-id]
+             #(+ % (seconds->hours (timers-core/elapsed-time timer)))))
```

Comment:
In a function below, the var `timers` represents a collection of timers, so I thought that aliasing a namespace as `timers` might be misleading/problematic.